base/types/gfsperrors/error.go

```diff
@@ -43,6 +46,23 @@ func init() {
 	})
 }
 
+func GetGfSpErr(err error) *GfSpError {
```

Comment:
what if the passed-in err is already a GfSpError?

Reply:
Never mind. The.Error() method will also return a parsable string, which will then be parsed to form a GfSpError